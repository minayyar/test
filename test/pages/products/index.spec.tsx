import { useRouter } from "next/router";
import { ROW_NUMBERS } from '@mocks/hooks';
import Products from '@pages/products/index';
import { render, screen } from '@test/utils';

jest.mock('@lib/hooks', () => require('@mocks/hooks'));
jest.mock('next/router', () => ({
    useRouter: jest.fn(),
}));

describe('Product List', () => {
    test('renders correctly', async () => {
        const router = { };
        useRouter.mockReturnValue(router);

        const { container } = render(<Products />);
        // Wait for table to be rendered
        await screen.findByRole('table');

        expect(container.firstChild).toMatchSnapshot();
    });

    test('renders a table with correct number of rows', async () => {
        const router = { };
        useRouter.mockReturnValue(router);

        render(<Products />);
        // Wait for table to be rendered
        const productsTable = await screen.findByRole('table');
        const rowsLength = screen.getAllByRole('row').length - 1; // Rows - 1 (table header)

        expect(productsTable).toBeDefined();
        expect(rowsLength).toEqual(ROW_NUMBERS);
    });
});
