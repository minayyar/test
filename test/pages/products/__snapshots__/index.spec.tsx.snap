// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List renders correctly 1`] = `
<div
  class="styled__StyledBox-sc-sj5f1m-0 dnUGcK styled__StyledPanel-sc-1h6ef3q-0 frQUTZ"
  id="products"
>
  <div
    aria-controls=":r0:"
    class="styled__StyledFlex-sc-yxkzj0-0 lkjaFf"
  >
    <div
      class="styled__StyledBox-sc-sj5f1m-0 bTHfrb styled__StyledFlexItem-sc-smjqtt-0 jjUmhN"
    >
      <p
        class="styled__StyledText-sc-tqnj75-5 bOpipu"
      >
        5 Products
      </p>
    </div>
    <div
      class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlexItem-sc-smjqtt-0 jjUmhN styled__StyledPaginationContainer-sc-1w0s9nk-0 gaoYmA"
    >
      <div
        aria-label="pagination"
        class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 dzErNZ"
        role="navigation"
      >
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlexItem-sc-smjqtt-0 gMwCkk"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledBox-sc-2535c9-0 eIBwbO"
          >
            <button
              aria-activedescendant=""
              aria-controls=":r1:-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              class="styled__StyledButton-sc-3yq204-0 htmsKm styled__StyledButton-sc-12shuvi-0 bZClQx"
              id=":r1:-toggle-button"
              role="button"
              tabindex="0"
              type="button"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                1 - 5 of 5
                <svg
                  aria-hidden="true"
                  class="base__StyledIcon-sc-a9u0e1-0 hNTNsv"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M8.71 11.71l2.59 2.59c.39.39 1.02.39 1.41 0l2.59-2.59c.63-.63.18-1.71-.71-1.71H9.41c-.89 0-1.33 1.08-.7 1.71z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="styled__StyledBox-sc-sj5f1m-0 epYaKJ"
              style="position: fixed; left: 0px; top: 0px; transform: translate(0px, 4px);"
            >
              <ul
                aria-labelledby=":r1:-label"
                class="styled__StyledList-sc-5w3dcq-0 bEnmuC"
                id=":r1:-menu"
                role="menu"
                tabindex="-1"
              />
            </div>
          </div>
        </div>
        <div
          class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlexItem-sc-smjqtt-0 gMwCkk"
        >
          <button
            class="styled__StyledButton-sc-3yq204-0 gcxeQP styled__StyledButton-sc-12shuvi-0 bZClQx"
            disabled=""
            type="button"
          >
            <span
              class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
            >
              <svg
                aria-labelledby=":r3:"
                class="base__StyledIcon-sc-a9u0e1-0 brjSIf"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
              >
                <title
                  id=":r3:"
                >
                  Previous page
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M14.71 6.71a.996.996 0 00-1.41 0L8.71 11.3a.996.996 0 000 1.41l4.59 4.59a.996.996 0 101.41-1.41L10.83 12l3.88-3.88c.39-.39.38-1.03 0-1.41z"
                />
              </svg>
            </span>
          </button>
          <button
            class="styled__StyledButton-sc-3yq204-0 gcxeQP styled__StyledButton-sc-12shuvi-0 bZClQx"
            disabled=""
            type="button"
          >
            <span
              class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
            >
              <svg
                aria-labelledby=":r4:"
                class="base__StyledIcon-sc-a9u0e1-0 brjSIf"
                fill="currentColor"
                height="24"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="24"
              >
                <title
                  id=":r4:"
                >
                  Next page
                </title>
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M9.29 6.71a.996.996 0 000 1.41L13.17 12l-3.88 3.88a.996.996 0 101.41 1.41l4.59-4.59a.996.996 0 000-1.41L10.7 6.7c-.38-.38-1.02-.38-1.41.01z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <table
    class="styled__StyledTable-sc-xns2ns-1 kbgftC"
    id=":r0:"
  >
    <thead
      class="styled__StyledTableHead-sc-1p8j1x-0"
    >
      <tr>
        <th
          class="styled__StyledTableHeaderCell-sc-1opnvtt-0 ljTCUO"
          id="header-cell-0"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 eYuALN styled__StyledFlex-sc-1opnvtt-2 kXIdGS"
          >
            Product name
          </div>
        </th>
        <th
          class="styled__StyledTableHeaderCell-sc-1opnvtt-0 ljTCUO"
          id="header-cell-1"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 eYuALN styled__StyledFlex-sc-1opnvtt-2 kXIdGS"
          >
            Stock
          </div>
        </th>
        <th
          class="styled__StyledTableHeaderCell-sc-1opnvtt-0 ljTCUO"
          id="header-cell-2"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 eYuALN styled__StyledFlex-sc-1opnvtt-2 kXIdGS"
          >
            Price
          </div>
        </th>
        <th
          class="styled__StyledTableHeaderCell-sc-1opnvtt-0 cDEKvS"
          id="header-cell-3"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledFlex-sc-hcvk8l-0 eYuALN styled__StyledFlex-sc-1opnvtt-2 jnetuN"
          >
            Action
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="styled__StyledTableBody-sc-1bwe2hn-0"
    >
      <tr
        class="styled__StyledTableRow-sc-xq0hr0-0 iXHitD"
      >
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <a
            href="/products/0"
          >
            <a
              class="styled__StyledLink-sc-n9mww9-0 haEQqw"
            >
              Product 0
            </a>
          </a>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <p
            class="styled__StyledSmall-sc-tqnj75-6 hpuiwL"
            color="danger"
          >
            0
          </p>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          $10.00
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledBox-sc-2535c9-0 eIBwbO"
          >
            <button
              aria-activedescendant=""
              aria-controls=":r9:-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              class="styled__StyledButton-sc-3yq204-0 gcxeQP bd-button"
              id=":r9:-toggle-button"
              role="button"
              tabindex="0"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                <svg
                  aria-hidden="true"
                  class="base__StyledIcon-sc-a9u0e1-0 PHcub"
                  color="secondary60"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="styled__StyledBox-sc-sj5f1m-0 epYaKJ"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <ul
                aria-labelledby=":r9:-label"
                class="styled__StyledList-sc-5w3dcq-0 bEnmuC"
                id=":r9:-menu"
                role="menu"
                tabindex="-1"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="styled__StyledTableRow-sc-xq0hr0-0 iXHitD"
      >
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <a
            href="/products/1"
          >
            <a
              class="styled__StyledLink-sc-n9mww9-0 haEQqw"
            >
              Product 1
            </a>
          </a>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <p
            class="styled__StyledSmall-sc-tqnj75-6 hpuiwL"
            color="danger"
          >
            0
          </p>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          $20.00
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledBox-sc-2535c9-0 eIBwbO"
          >
            <button
              aria-activedescendant=""
              aria-controls=":rb:-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              class="styled__StyledButton-sc-3yq204-0 gcxeQP bd-button"
              id=":rb:-toggle-button"
              role="button"
              tabindex="0"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                <svg
                  aria-hidden="true"
                  class="base__StyledIcon-sc-a9u0e1-0 PHcub"
                  color="secondary60"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="styled__StyledBox-sc-sj5f1m-0 epYaKJ"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <ul
                aria-labelledby=":rb:-label"
                class="styled__StyledList-sc-5w3dcq-0 bEnmuC"
                id=":rb:-menu"
                role="menu"
                tabindex="-1"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="styled__StyledTableRow-sc-xq0hr0-0 iXHitD"
      >
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <a
            href="/products/2"
          >
            <a
              class="styled__StyledLink-sc-n9mww9-0 haEQqw"
            >
              Product 2
            </a>
          </a>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <p
            class="styled__StyledSmall-sc-tqnj75-6 hpuiwL"
            color="danger"
          >
            0
          </p>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          $30.00
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledBox-sc-2535c9-0 eIBwbO"
          >
            <button
              aria-activedescendant=""
              aria-controls=":rd:-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              class="styled__StyledButton-sc-3yq204-0 gcxeQP bd-button"
              id=":rd:-toggle-button"
              role="button"
              tabindex="0"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                <svg
                  aria-hidden="true"
                  class="base__StyledIcon-sc-a9u0e1-0 PHcub"
                  color="secondary60"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="styled__StyledBox-sc-sj5f1m-0 epYaKJ"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <ul
                aria-labelledby=":rd:-label"
                class="styled__StyledList-sc-5w3dcq-0 bEnmuC"
                id=":rd:-menu"
                role="menu"
                tabindex="-1"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="styled__StyledTableRow-sc-xq0hr0-0 iXHitD"
      >
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <a
            href="/products/3"
          >
            <a
              class="styled__StyledLink-sc-n9mww9-0 haEQqw"
            >
              Product 3
            </a>
          </a>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <p
            class="styled__StyledSmall-sc-tqnj75-6 hpuiwL"
            color="danger"
          >
            0
          </p>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          $40.00
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledBox-sc-2535c9-0 eIBwbO"
          >
            <button
              aria-activedescendant=""
              aria-controls=":rf:-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              class="styled__StyledButton-sc-3yq204-0 gcxeQP bd-button"
              id=":rf:-toggle-button"
              role="button"
              tabindex="0"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                <svg
                  aria-hidden="true"
                  class="base__StyledIcon-sc-a9u0e1-0 PHcub"
                  color="secondary60"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="styled__StyledBox-sc-sj5f1m-0 epYaKJ"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <ul
                aria-labelledby=":rf:-label"
                class="styled__StyledList-sc-5w3dcq-0 bEnmuC"
                id=":rf:-menu"
                role="menu"
                tabindex="-1"
              />
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="styled__StyledTableRow-sc-xq0hr0-0 iXHitD"
      >
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <a
            href="/products/4"
          >
            <a
              class="styled__StyledLink-sc-n9mww9-0 haEQqw"
            >
              Product 4
            </a>
          </a>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <p
            class="styled__StyledSmall-sc-tqnj75-6 hpuiwL"
            color="danger"
          >
            0
          </p>
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          $50.00
        </td>
        <td
          class="styled__StyledTableDataCell-sc-16y6nzi-0 eLbxfq"
        >
          <div
            class="styled__StyledBox-sc-sj5f1m-0 BRgwK styled__StyledBox-sc-2535c9-0 eIBwbO"
          >
            <button
              aria-activedescendant=""
              aria-controls=":rh:-menu"
              aria-expanded="false"
              aria-haspopup="menu"
              class="styled__StyledButton-sc-3yq204-0 gcxeQP bd-button"
              id=":rh:-toggle-button"
              role="button"
              tabindex="0"
            >
              <span
                class="styled__ContentWrapper-sc-3yq204-1 ihVQNH"
              >
                <svg
                  aria-hidden="true"
                  class="base__StyledIcon-sc-a9u0e1-0 PHcub"
                  color="secondary60"
                  fill="currentColor"
                  height="24"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                  />
                </svg>
              </span>
            </button>
            <div
              class="styled__StyledBox-sc-sj5f1m-0 epYaKJ"
              style="position: absolute; left: 0px; top: 0px;"
            >
              <ul
                aria-labelledby=":rh:-label"
                class="styled__StyledList-sc-5w3dcq-0 bEnmuC"
                id=":rh:-menu"
                role="menu"
                tabindex="-1"
              />
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
